### PR TITLE
Fix get-updates and activate to use GUID in addition to device-id

### PR DESCRIPTION
**Pull request by djcampello**
Also merge fu_engine_get_device and fu_engine_get_device_by_id

Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ X ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-engine.c [hunk 1]
```diff
@@ -2399,19 +2399,17 @@ fu_engine_get_plugins (FuEngine *self)
...
- *
- * Returns the device from the engine or NULL if not found
+ * @device_id: A device ID
+ * @error: A #GError, or %NULL
  *
- * Returns: (transfer full): a #FuDevice
+ * Gets a specific device.
  *
- * Since: 1.5.0
+ * Returns: (transfer full): a device, or %NULL if not found
```
Comment (superm1) [inline src/fu-engine.c:2409]: This should still have the clause mentioning when it was introduced.  IE:
` * Since: 1.5.0	`

**Review by superm1 — APPROVED** [review]
just a small nit, otherwise LGTM

File: src/fu-engine.c (hunk 1 above)
Comment (djcampello) [reply src/fu-engine.c:2409]: This function has been there from before.... I just eliminated the _by_id that was going to be introduced on 1.5.0

Comment (superm1) [reply src/fu-engine.c:2409]: Oh I see... OK.
LGTM then

**Pull request merged**